### ISSUE TITLE
How to convert the YOLOv8 model into tfjs

### ISSUE THREAD

**Issue by jef-way** [Hyuto/yolov8-tfjs#11]
I would like to ask how your YOLOv8 model converts to tfjs. I trained a model myself, but the teacher reported an error when converting to tfjsI would like to ask how your YOLOv8 model converts to tfjs. I trained a model myself, but the teacher reported an error when converting to tfjs

**Comment by Hyuto** [comment]
Hello, Exporting YOLOv8 model to TFJS can be done using `export` method, as you can see on the documentation [Export Usage Examples](https://docs.ultralytics.com/modes/export/#usage-examples) with `format="tfjs"`.

**Example**

```python
from ultralytics import YOLO

model = YOLO('PATH/TO/CHECKPOINT.pt')  # load a custom trained model

model.export(format='tfjs') # Export the model
```

I recommend to upgrade `ultralitics` library to the latest version, maybe your teacher isn't using latest version of `ultralitics` library, because the old version of the library can't export to `tfjs` format yet.

**Comment by jef-way** [comment]
Hello, the inputs [0]. shape of your YOLOv8 model is [1, 640, 640, 3], while the inputs [0]. shape of my self trained model is [1, 3, 640, 640]. Why is this?
![微信截图_20240116114211](https://github.com/Hyuto/yolov8-tfjs/assets/2706691/e73b6d62-12e9-4e19-a9ed-92cb49ac8b9c)
![微信截图_20240116114148](https://github.com/Hyuto/yolov8-tfjs/assets/2706691/1f8e0297-2224-4606-ade7-40450a67823e)

**Comment by Hyuto** [comment]
Yes, the input shape of all TF model should be with [batch, height, width, channels] format, I don't know why your model is using [batch, channels, height, width] format which is usually used in pytorch or onnx model. Do you use the latest version of `ultralitics` to convert your custom trained model to tfjs?

**Comment by jef-way** [comment]
The error below is reported when exporting a model trained with the latest version of Ultralitics
![296646799-734e28a4-cbf1-45e6-9000-e7aa108d81ee](https://github.com/Hyuto/yolov8-tfjs/assets/2706691/762bc56a-bc1c-47ca-abe5-a9eb00862ff3)
So I took the exported onnx and first converted it to SavedModel, and then used tensorflowjs_ Convert to tfjs

**Comment by Hyuto** [comment]
Well that's normal because YOLOv8 ONNX model input shape is [b, c, h, w] format, and you just convert it raw. I suggest to `transpose` the input image tersor before forwarding it to the net, to avoid error

**Example**

```js
...
const [input, xRatio, yRatio] = preprocess(source, modelWidth, modelHeight); // preprocess image
const transInput = input.transpose([0, 3, 1, 2]) // make it [b, c, h, w] from [b, h, w, c] # default tensorflow
...
```

**Comment by MaryamBoneh** [comment]
hi
i want convert yolov8m-pose.pt to tfjs format. i used yolo model.export and i get yolov8m-pose_float32.tflite and saved_model.pb and... not like what you get from export .pt to .tfjs.
